### DISCUSSION TITLE
Trying to install Exchange EWS Connector (Exchange Web Services) APP - Missing SOAP

### DISCUSSION THREAD

**Discussion by JSChasle** [nextcloud/all-in-one#3794] [Ideas] [answered]
I was wondering if it would be easy to add the soap library so I can install the EWS Connector. Am getting message the library soap is not available. Am on version 27.1.3.

Thanks

**Comment by szaimen** [comment, marked as answer]
Hi, see https://github.com/nextcloud/all-in-one#how-to-add-php-extensions-permanently-to-the-nextcloud-container

**Reply by JSChasle** [reply]
Sorry for my ignorance, but where is the master container located? I would need to nano the run script. and then add before the nextcloud/all-in-one:latest I would add the line --env NEXTCLOUD_ADDITIONAL_PHP_EXTENSIONS="soap"

**Reply by szaimen** [reply]
> but where is the master container located?

It runs as container on your server

**Reply by JSChasle** [reply]
So if I understand correctly since soap does not exist on https://pecl.php.net/package-search.php?pkg_name=soap I can`t install the package (thanks for your patience)

And I can see the docker mastercontainer running (using docker ps command) but still ain't sure how I would go to edit the container to add the extention
